### PR TITLE
[CMakeV1] Migrated legacy tests

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
**Task name**: CMakeV1

**Description**: Migrated legacy tests to 

**Documentation changes required:** No

**Added unit tests:** Yes

**Attached related issue:** No

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/CMakeV1/Tests/L0.ts [hunk 1]
```diff
@@ -1,16 +1,61 @@
...
+        const expectedErr: string = '/usr/local/bin/cmake failed with return code: 50';
+        assert(runner.stdOutContained(expectedErr), 'should have printed: ' + expectedErr);
+        assert(runner.failed, 'task should have failed');
+
+        done();
+    })
+	
+	it('errors if cwd not set', (done: Mocha.Done) => {
+		
+		let testPath = path.join(__dirname, 'L0NoCwdSet.js');
```
Comment (alexander-smolyakov) [inline Tasks/CMakeV1/Tests/L0.ts:49]: Could you please fix the code formatting?

File: Tasks/CMakeV1/Tests/L0.ts [hunk 2]
```diff
@@ -1,16 +1,61 @@
...
+        assert(runner.ran('/usr/local/bin/cmake ..'), 'it should have run cmake');
+        assert(runner.invokedToolCount == 1, 'should have only run cmake');
+        assert(runner.stderr.length == 0, 'should not have written to stderr');
+        assert(runner.succeeded, 'task should have succeeded');
+
+        done()
+    })	
+
+	it('fails if cmake fails', (done: Mocha.Done) => {
+		let testPath = path.join(__dirname, 'L0ShouldFail.js');
```
Comment (alexander-smolyakov) [inline Tasks/CMakeV1/Tests/L0.ts:32]: code formatting

File: Tasks/CMakeV1/Tests/L0.ts [hunk 3]
```diff
@@ -1,16 +1,61 @@
...
+        assert(runner.stdOutContained(expectedErr), 'should have said: ' + expectedErr);
+        assert.strictEqual(runner.invokedToolCount, 0, 'should exit before running CMake');
 
-    it('Does a basic hello world test', function(done: Mocha.Done) {
-        // TODO - add real tests
         done();
-    });
-});
+    })
+});
```
Comment (alexander-smolyakov) [inline Tasks/CMakeV1/Tests/L0.ts:61]: Nit: EOL

File: Tasks/CMakeV1/Tests/L0NoCwdSet.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "which": {
+        "cmake": "/usr/local/bin/cmake"
+    },
+    "checkPath": {
+        "/usr/local/bin/cmake": true
+    }
+}
+tmr.setAnswers(answers);
+
+tmr.run();
```
Comment (alexander-smolyakov) [inline Tasks/CMakeV1/Tests/L0NoCwdSet.ts:18]: Nit: EOL

File: Tasks/CMakeV1/Tests/L0RunInCwd.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    "exec": {
+        "/usr/local/bin/cmake ..": {
+            "code": 0,
+            "stdout": "cmake output here"
+        }
+    }
+}
+tmr.setAnswers(answers);
+
+tmr.run();
```
Comment (alexander-smolyakov) [inline Tasks/CMakeV1/Tests/L0RunInCwd.ts:29]: Nit: EOL

File: Tasks/CMakeV1/Tests/L0ShouldFail.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        "/usr/local/bin/cmake ..": {
+            "code": 50,
+            "stdout": "cmake output here",
+            "stderr": "cmake failed with this output"
+        }
+    }
+}
+tmr.setAnswers(answers);
+
+tmr.run();
```
Comment (alexander-smolyakov) [inline Tasks/CMakeV1/Tests/L0ShouldFail.ts:30]: Nit: EOL

**Review by alexander-smolyakov — APPROVED** [review]
LGTM, thanks!

File: Tasks/CMakeV1/Tests/L0.ts [hunk 7]
```diff
@@ -1,16 +1,59 @@
...
 import assert = require('assert');
 import path = require('path');
+import { MockTestRunner } from 'azure-pipelines-task-lib/mock-test';
 
-describe('CMakeV1 Suite', function () {
-    before(() => {
+describe('CMake Suite', function () {
+    this.timeout(parseInt(process.env.TASK_TEST_TIMEOUT) || 20000);
+
+    before((done) => {
```
Comment (anatolybolshakov) [inline Tasks/CMakeV1/Tests/L0.ts:8]: Do we really need this section - since it's empty?

File: Tasks/CMakeV1/Tests/L0.ts [hunk 8]
```diff
@@ -1,16 +1,59 @@
...
+        done();
     });
 
-    after(() => {
+    after(function () {
```
Comment (anatolybolshakov) [inline Tasks/CMakeV1/Tests/L0.ts:12]: Same for after

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks! Could you take a look at comments also?

**Review by DaniilShmelev — APPROVED** [review]

**Pull request merged**